mmv1/third_party/terraform/services/composer/resource_composer_environment.go.erb

```diff
@@ -1548,10 +1554,14 @@ func flattenComposerEnvironmentConfig(envCfg *composer.EnvironmentConfig) interf
 	transformed["airflow_uri"] = envCfg.AirflowUri
 	transformed["node_config"] = flattenComposerEnvironmentConfigNodeConfig(envCfg.NodeConfig)
 	transformed["software_config"] = flattenComposerEnvironmentConfigSoftwareConfig(envCfg.SoftwareConfig)
-	transformed["private_environment_config"] = flattenComposerEnvironmentConfigPrivateEnvironmentConfig(envCfg.PrivateEnvironmentConfig)
+	if !isComposer3(envCfg){
```

Comment:
Why is this being excluded for composer2? (And the same question for all the other places where this is happening.) This will change what's stored in state for the resource.

Reply:
Intention here was to have these two fields ("enable_private_environment" and "enable_private_builds_only") under "config" instead of nested in "private_environment_config". But these fields still map to privateEnvironmentConfig in API. So I added two different conversions base on Composer version.